atomate/qchem/firetasks/fragmenter.py

```diff
@@ -0,0 +1,227 @@
+                    "formula_pretty": {
+                        "$in": unique_formulae
+                    }
+                }, {
+                    "formula_pretty": 1,
+                    "output.initial_molecule": 1
+                }))
+
+        # build the list of new fireworks
+        new_FWs = build_new_FWs(unique_molecules, all_relevant_docs,
+                                self.get("max_cores", 32),
+                                self.get("qchem_input_params", {}))
+
+        return FWAction(additions=new_FWs)
+
+
+def edges_from_babel(molecule):
+    babel_mol = BabelMolAdaptor(molecule).openbabel_mol
+    edges = []
+    for obbond in ob.OBMolBondIter(babel_mol):
+        edges += [[obbond.GetBeginAtomIdx() - 1, obbond.GetEndAtomIdx() - 1]]
+    return edges
+
+
+def build_MoleculeGraph(molecule, edges=None):
+    if edges == None:
+        edges = edges_from_babel(molecule)
+    mol_graph = MoleculeGraph.with_empty_graph(molecule)
+    for edge in edges:
+        mol_graph.add_edge(edge[0], edge[1])
+    mol_graph.graph = mol_graph.graph.to_undirected()
+    species = {}
+    coords = {}
+    for node in mol_graph.graph:
+        species[node] = mol_graph.molecule[node].specie.symbol
+        coords[node] = mol_graph.molecule[node].coords
+    nx.set_node_attributes(mol_graph.graph, species, "specie")
+    nx.set_node_attributes(mol_graph.graph, coords, "coords")
+    return mol_graph
+
+
+def build_unique_fragments(mol_graph):
+    # find all possible fragments, aka connected induced subgraphs
+    all_fragments = []
+    for ii in range(1, len(mol_graph.molecule)):
+        for combination in combinations(mol_graph.graph.nodes, ii):
+            subgraph = nx.subgraph(mol_graph.graph, combination)
+            if nx.is_connected(subgraph):
+                all_fragments.append(subgraph)
+
+    # narrow to all unique fragments using graph isomorphism
+    unique_fragments = []
+    for fragment in all_fragments:
+        if not [is_isomorphic(fragment, f)
+                for f in unique_fragments].count(True) >= 1:
+            unique_fragments.append(fragment)
+    return unique_fragments
+
+
+def build_unique_molecules(unique_fragments, orig_charge):
+    unique_molecules = []
+    for fragment in unique_fragments:
+        species = [fragment.node[ii]["specie"] for ii in fragment.nodes]
+        coords = [fragment.node[ii]["coords"] for ii in fragment.nodes]
+        unique_molecule0 = Molecule(
+            species=species, coords=coords, charge=orig_charge)
+        unique_molecule1 = Molecule(
+            species=species, coords=coords, charge=orig_charge + 1)
+        unique_molecule2 = Molecule(
+            species=species, coords=coords, charge=orig_charge - 1)
+        unique_molecules.append(unique_molecule0)
+        unique_molecules.append(unique_molecule1)
+        unique_molecules.append(unique_molecule2)
+    return unique_molecules
+
+
+def _node_match(node, othernode):
```

Comment:
do these one-liners need a specific function? The code seems more readable to just put the line directly where it is needed versus having to follow the code down to some private function and then figure out what's in that private function. The code itself is clear that we are checking the specie of the node for equivalence. The name _node_match is not very clear...

If there is a specific Node object you can also override the __eq__ function to check that the corresponding specie is the same, not sure if this makes sense. But then you can just do node == othernode which is very compact and clear

Reply:
Improved and moved to pymatgen